client/src/app/App.js

```diff
@@ -221,17 +221,24 @@ export class App extends PureComponent {
 
     // skip unchanged
     if (!(lastModified > tabLastModified)) {
-      return;
+      return tab;
     }
 
     const answer = await this.showDialog(getContentChangedDialog());
 
     if (answer === 'ok') {
-      const updatedFile = await fileSystem.readFile(file.path);
+      const updatedFile = await fileSystem.readFile(file);
```

Comment:
[`fileSystem.readFile` ](https://github.com/camunda/camunda-modeler/blob/master/client/src/remote/FileSystem.js#L20) is intended to only get a file's path instead of the whole file:

```javascript
/**
 * Read file.
 *
 * @param {String} filePath - Filepath.
 * @param {Object} [options] - Options.
 * @param {String} [options.encoding] - Encoding.
 *
 * @returns {Promise}
 */
readFile(filePath, options = {}) {
  return this.backend.send('file:read', filePath, options);
}
```

That's why the tab content does not get updated properly atm. It remains the old state because the correct file does not get fetched:

![image](https://user-images.githubusercontent.com/9433996/53473363-83947600-3a6a-11e9-8276-afe20ef9efcf.png)

Reply:
I'll test my fix locally.

Reply:
Fixed via https://github.com/camunda/camunda-modeler/pull/1277/commits/702b0c6ef359b66efe6d6b11b7a3a933661d9d59.